db/event_manager.go

```diff
@@ -12,11 +13,37 @@ import (
 // eventChannel to minimize time spent blocking whatever process is raising the event.
 // The event queue worker goroutine works the event channel and sends events to the appropriate handlers
 type EventManager struct {
-	activeEventTypes   map[EventType]bool
-	eventHandlers      map[EventType][]EventHandler
-	asyncEventChannel  chan Event
-	activeCountChannel chan bool
-	waitTime           int
+	activeEventTypes       map[EventType]bool
+	eventHandlers          map[EventType][]EventHandler
+	asyncEventChannel      chan Event
+	activeCountChannel     chan bool
+	waitTime               int
+	eventsProcessedSuccess int64
+	eventsProcessedFail    int64
+}
+
+func (em *EventManager) GetEventsProcessedSuccess() int64 {
+	return atomic.LoadInt64(&em.eventsProcessedSuccess)
+}
+
+func (em *EventManager) IncrementEventsProcessedSuccess() int64 {
+	return atomic.AddInt64(&em.eventsProcessedSuccess, 1)
+}
+
+func (em *EventManager) DecrementEventsProcessedSuccess() int64 {
```

Comment:
The decrement helpers aren't used, so I'd probably remove until there's a real use case for them.  (even at that point, I'd probably suggest modifying IncrementEventsProcessed[...] to accept an int as input).

Reply:
Opted to add the delta input as suggested